Add cached tokens to Unified API response

jaybcee opened the pull request:
This PR adds support for cached tokens in the Unified Chat Completion API response, allowing users to track prompt caching from EIS and OpenAI services.

### Testing
- **`OpenAiUnifiedStreamingProcessorTests.java`**: Added comprehensive tests for both scenarios (with and without cached tokens)
- Tests verify correct parsing and backward compatibility

## Response Format

When cached tokens are present, the response includes:
```json
{
  "completion_tokens": 150,
  "prompt_tokens": 55,
  "total_tokens": 205,
  "prompt_tokens_details": {
    "cached_tokens": 20
  }
}
```

The `prompt_tokens_details` object is optional and only appears when cached token information is available from the EIS service, following the OpenAI specification.

I did not implement this for non open-ai providers. We don't really need this field at the moment, it just helps with O11y tools like Phoenix.

jaybcee committed:
Add cached tokens to Unified API response

Commit:
[CI] Auto commit changes from spotless

jaybcee committed:
Parse cached tokens from EIS/OpenAI responses in unified API

- Updated OpenAiUnifiedStreamingProcessor to parse optional cached_tokens from prompt_tokens_details
- Added ConstructingObjectParser for prompt_tokens_details nested object
- Added tests for both cached tokens present and absent scenarios
- Maintains backward compatibility with responses without cached tokens

Commit:
[CI] Auto commit changes from spotless

jaybcee committed:
Update test expectations to include cached_tokens in output

- Updated OpenAiServiceTests.testUnifiedCompletionInfer to expect cached_tokens:0 in response
- Updated HuggingFaceServiceTests.testUnifiedCompletionInfer to expect cached_tokens:0 in response
- Including cached_tokens:0 provides meaningful information (caching available but not used)
- Distinguishes from null (no caching information available)

jaybcee committed:
Merge branch 'main' into add-cached-tokens-unified-api

jaybcee committed:
Merge branch 'main' into add-cached-tokens-unified-api

jaybcee committed:
Update docs/changelog/136412.yaml

jonathan-buttner reviewed (commented):
Looking good, left a few comments about the transport version check.

jaybcee committed:
Merge branch 'main' into add-cached-tokens-unified-api

jaybcee committed:
Address comments: Add transport version checks for cached tokens

Wrap cachedTokens read/write operations in transport version checks to ensure
backward compatibility with older nodes that haven't been upgraded yet.

Commit:
[CI] Update transport version definitions

jaybcee committed:
Remove CSV for Transports such that CI creates a new one

jaybcee committed:
Restore 9.3.csv transport version definitions

jaybcee committed:
Update 9.3.csv to match main branch

jaybcee committed:
Merge branch 'main' into add-cached-tokens-unified-api

Commit:
[CI] Update transport version definitions
